File: Lily_Can_Do_It.cpp
/*
||-----------------------------------||
||        Rahat Khan Pathan          ||
||         Programming Hero          ||
||-----------------------------------||
*/
#include<bits/stdc++.h>
#define ll long long int
#define fast ios_base::sync_with_stdio(0); cin.tie(NULL);
#define yes cout<<"YES"<<endl
#define no cout<<"NO"<<endl
using namespace std;
int main()
{
    fast;
    ll t;
    cin>>t;
    while(t--)
    {
        ll n;
        cin>>n;
        ll e=0,od=0;
        for(ll i=0;i<n;i++)
        {
            ll a;
            cin>>a;
            if(a%2==0) e++;
            else od++;
        }
        if(e && od) yes;
        else no;
    }
    return 0;
}

File: Lily_Can_Do_It.py
t=int(input())
while t>0:
    t-=1
    n=int(input())
    l=list(map(int, input().split()))
    o=0
    e=0
    for a in l:
        if a%2==0:
            e+=1
        else:
            o+=1
    if e>0 and o>0:
        print("YES")
    else:
        print("NO")
